public/app/plugins/panel/heatmap-new/fields.ts

```diff
@@ -59,6 +60,9 @@ export function prepareHeatmapData(
   // detect a frame-per-bucket heatmap frame
   // TODO: improve heuristic? infer from fields[1].labels.le === '+Inf' ?
   if (frames[0].meta?.custom?.resultType === 'matrix' && frames.some((f) => f.name?.startsWith('+Inf'))) {
+    // already done by the Prometheus datasource frontend
+    //frames = prepBucketFrames(frames);
```

Comment:
IIUC -- this is removing the de-accumulation

Lets see if this works, if it does -- we should cleanup and add tests etc

Reply:
sounds good. longer term i was planning on re-using this fn in the prom datasource instead of what's currently there since this one is a bit more efficient and cleaner.